### ISSUE TITLE
Query: AQL: missing variable #55 (54) for node #4 (CalculationNode) while planning registers

### ISSUE THREAD

**Issue by davechallis** [arangodb/arangodb#9660]
## My Environment

* __ArangoDB Version__:       3.5 RC7
* __Storage Engine__:            RocksDB
* __Deployment Mode__:       Single Server
* __Deployment Strategy__:   ArangoDB Starter in Docker

## Component, Query & Data

__Affected feature__:

AQL query using web interface

__AQL query (if applicable)__:

Apologies for the complex query, but have been struggling to find a simpler version which also triggers the error.

Basically it does a text search to find some nodes in a graph, then traverses the graph upwards to find the root of each tree, then tries to aggregate on the tree root's name, and the maximum scoring node that was found with a text search:

```
LET tokens = TOKENS(@search, "text_en")
FOR a IN nameSearchView
  SEARCH ANALYZER(STARTS_WITH(a.name, tokens[0]) OR a.name IN tokens, "text_en")
  LET score = BM25(a, 20.0, 0.01) + 1.0
  SORT score DESC
  LIMIT 50
  LET root = LENGTH(FOR v,e,p IN 1..5 OUTBOUND a belongsTo RETURN e) == 0 ? a : FIRST(
    FOR v1, e1, p1 IN 1..5 OUTBOUND a belongsTo
    FILTER LENGTH(FOR v2,e2,p2 IN 1..5 OUTBOUND v1 belongsTo RETURN e2) == 0
    RETURN DISTINCT v1
  )
  LET rootScore = {"root": root, "score": score}
  COLLECT parent = rootScore.root.name AGGREGATE maxScore = MAX(rootScore.score)
  SORT maxScore DESC
  RETURN {"name": parent, "score": maxScore}
```

Part of the problem seems to be related to:

```
LET score = BM25(a, 20.0, 0.01) + 1.0
```

removing the `+ 1.0` and running with:

```
LET score = BM25(a, 20.0, 0.01)
```

and the query executes without any problem (I'm actually doing something more complicated with modifying the score, but any modification seems to cause the error).

Likewise, it also goes away if I remove the aggregation.

__Problem__:

Executing the above query gives the following error in the web UI:

```
Query: AQL: missing variable #55 (54) for node #4 (CalculationNode) while planning registers (exception location: /work/ArangoDB/arangod/Aql/ExecutionNode.cpp:1101). Please report this error to arangodb.com (while optimizing plan) (exception location: /work/ArangoDB/arangod/RestHandler/RestCursorHandler.cpp:279). Please report this error to arangodb.com
```

**Comment by gnusi** [comment]
Hi @davechallis, thank you for submitting the issue. I can confirm that's a bug in `3.5RC7`.
You can expect it will be fixed in `3.5.1`

**Comment by gnusi** [comment]
@davechallis FYI
fixed in `devel` and `3.5.1`

**Comment by Simran-B** [comment]
v3.5.1 was released, so closing.
